src/app/contextBroker/contextBroker.cpp

```diff
@@ -772,12 +777,14 @@ static const char* validLogLevels[] =
 
 
 #define API_V2                                                                                         \
-  { "GET",    EPS,          EPS_COMPS_V2,         ENT_COMPS_WORD,          entryPointsTreat         }, \
-  { "*",      EPS,          EPS_COMPS_V2,         ENT_COMPS_WORD,          badVerbGetOnly           }, \
+  { "GET",     EPS,          EPS_COMPS_V2,        ENT_COMPS_WORD,          entryPointsTreat         }, \
+  { "OPTIONS", EPS,          EPS_COMPS_V2,        ENT_COMPS_WORD,          optionsGetOnly           }, \
+  { "*",       EPS,          EPS_COMPS_V2,        ENT_COMPS_WORD,          badVerbGetOnly           }, \
```

Comment:
Up to know badVerbGetOnly generates something like this:

```
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: GET
```

however, after this PR, I understand this should be:

```
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: GET, OPTIONS
```
Is that correct?

(Similar comment for `badVerbGetPostOnly` function below)

Reply:
I agree but this should only be the case when CORS is enabled. Maybe we can add add a new logic in `badVerb...` handlers to check if CORS is enabled and if so add OPTIONS to the outputs.

Which brings us to your question earlier (which we skipped and got buried under other comments). What if CORS is not enabled and we receive an OPTIONS request anyway? These kinds of requests will not be treated with the ´badVerb...´ handlers since we already have the OPTIONS in the service vectors.

> Also related with this, I understand that OPTIONS processing make only sense in the case of CORS functionality (at least by the time being). Thus, the OPTIONS routes should be enabled only when CORS is activated. In other words, these restServiceV[] vector should include these entries only if -enableCORS (or whicheer name we use at the end for the CLI parameter) is being used.
> We did in the past a similar "trick" with the -ngsi9 switch (which disables all the NGSI9 routes in the V1 API). @kzangeli do you remember how we did that time? Otherwise, maybe we should look to old versions of the contextBroker.app (go to tag 0.24.0 in github), as -ngsi9 was deprecated and removed time ago.

Reply:
With regards to:

> Maybe we can add add a new logic in badVerb... handlers to check if CORS is enabled and if so add OPTIONS to the outputs.

Correct. It should be just a matter of checking `-enableCORS` (or whichever name we use at the end for the CLI parameter) has been provided or not, adjusting the list to include in the `Allow` header accordingly.

> What if CORS is not enabled and we receive an OPTIONS request anyway?

That should be considered a bad verb case and, if `restServiceV[]` is corrrectly set (as I outline in my previous comment, see 0.24.0 code), the request will be reouted to the right bad verb handler.

Reply:
Actually, we will need some logic to make this work. A conditional that looks at the -cors switch when looking up the services.
This is my backyard, I can take a look at it soon. If you want, just ignore this and I'll fix it later. Soon

Reply:
I am sorry for the spam @fgalan but I need a confirmation. I checked the tag you mentioned and I have some questions about the details. 

So basically in the main function, we chose which rest service vector to use depending on the switch.

`RestService* rsP       = (ngsi9Only == true)? restServiceNgsi9 : restServiceV;`

So I will create another vector called `restServiceCORS` and choose this or the original depending on the switch. And instead of having `API_V2` inside this vector, I can use a duplicate of the whole `API_V2` as `API_V2_CORS` which also includes the OPTIONS handlers. 

This would work but my question is do we really need to duplicate the whole `API_V2` definition? Is it possible to define `API_V2_CORS` with only OPTIONS entries like below: 

    { "OPTIONS", EPS,          EPS_COMPS_V2,        ENT_COMPS_WORD,          optionsGetOnly           },
    { "OPTIONS", ENT,          ENT_COMPS_V2,        ENT_COMPS_WORD,          optionsGetPostOnly       },
    ...

and create the `restServiceCORS` vector as below:

    RestService restServiceV[] =
    {
      API_V2,
      API_V2_CORS,

      REGISTRY_STANDARD_REQUESTS_V0,
      ...

Would this work or the sequence of the entries are important?

Reply:
With regards to bad verd including the right set of verbs for `Allows` I agree in deferring to a next PR.

With regards disable OPTIONS routes when `-enableCORS` (or whichever name we use at the end for the CLI parameter) is not used, I'd like to see it in this PR. Orion users needs to control if they want cors-stuff included (and the potential security risks it brings ;) or not when they start the CB process.

Reply:
> Would this work or the sequence of the entries are important?

I'd say that the sequence of entries doesn't mind (*)

However, let's wait for @kzangeli feedback on this, which knows how this works better than me :)

(*) At least at this point... in the future, maybe some kind of optimization based on the frecuency of each operation could be implemented, but this is another history and outside the scope this PR).

Reply:
We will have a look at the order some day.
Lots of time can be saved putting more frequent requests in the beginning.

But, forget about that for now, I will change a lot of things in main.cpp for this.
Longer lines, as it was before. 
Right now it is very hard to understand these vectors

Reply:
And, about replicating the entire vector of services for -cors, I like it !
Didn't think about that (obvious) option !!!

Reply:
Fixed in 543c64b